lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitEvaluator.cs

```diff
@@ -0,0 +1,247 @@
+            {
+                Logger.Log(Level.Info, "RunningTask: " + value.Id);
+                string msg = value.Id.Equals(TaskId + "1") ? FailSignal : SuccSignal;
+                value.Send(Encoding.UTF8.GetBytes(msg));
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, CompletedTaskValidationMessage + ". Task completed: " + value.Id);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(ICompletedEvaluator value)
+            {
+                Logger.Log(Level.Info, CompletedEvaluatorValidationMessage + ". Evaluator completed: " + value.Id);
+            }
+
+            public void OnNext(IFailedEvaluator value)
+            {
+                Logger.Log(Level.Info, FailedEvaluatorMessage + ". Evaluator failed: " + value.Id + "  FailedTaskId: " + value.FailedTask.Value.Id);
+                Assert.NotNull(value.FailedTask);
+                Assert.NotNull(value.FailedTask.Value);
+                Assert.NotNull(value.FailedTask.Value.Id);
+                Assert.True(value.FailedTask.Value.Id.Equals(TaskId + "1") || value.FailedTask.Value.Id.Equals(TaskId + "2"));
+                Assert.Equal(value.FailedContexts.Count, 1);
+                Assert.True(value.FailedContexts[0].Id.Equals(ContextId + "1") || value.FailedContexts[0].Id.Equals(ContextId + "2"));
+
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IFailedContext value)
+            {
+                throw new Exception("Did not expect Failed Context.");
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                throw new Exception("Did not expect Failed Task.");
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private sealed class FailEvaluatorTask : ITask, IDriverMessageHandler, IObserver<ICloseEvent>
+        {
+            private static readonly Logger Logger = Logger.GetLogger(typeof(FailEvaluatorTask));
+
+            private readonly CountdownEvent _countdownEvent = new CountdownEvent(1);
+            private string _message;
+
+            [Inject]
+            private FailEvaluatorTask()
+            {
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                Logger.Log(Level.Info, "Hello in FailEvaluatorTask");
+                _countdownEvent.Wait();
+
+                if (_message.Equals(TestResubmitEvaluator.FailSignal))
+                {
+                    Environment.Exit(1);
+                }
+
+                // To make sure failed evaluator/context is submitted before other tasks are completed. 
+                Thread.Sleep(2000);
```

Comment:
I submitted two at beginning, one fails one successful. If the successful one returns too quick before the 3rd one is submitted, everything will be shut down. 
I could change the logic to not to dispose the successful context until the 3rd task is submitted. I thought about it, just more code for a simple testing.